packages/docs/docs/advanced/custom-components.mdx

```diff
@@ -0,0 +1,126 @@
+---
+slug: /custom-components
+---
+
+import AnimationPlayer from '@site/src/components/AnimationPlayer';
+import CodeBlock from '@theme/CodeBlock';
+import source from '!!raw-loader!@motion-canvas/examples/src/scenes/components-switch';
+
+# Components
+
+<AnimationPlayer name="components" banner />
+
+Components are reusable parts of code such as `Rect` and `Circle` which can be
+reused throughout scenes. To use a component in a scene, add it to the view and
+provide arguments to the component.
+
+```tsx
+<Switch
+    initialState={false}
+/>
+```
+To define what arguments a component will take in, first define an interface.
+All properties of the interface must be wrapped in `SignalValue<>` as such:
+```tsx
+//  You can extend an existing props interface
+//  such as LayoutProps, ShapeProps or NodeProps to
+//  include their properties alongside the ones you 
+//  define
+
+export interface SwitchProps extends NodeProps {
+    initialState?: SignalValue<boolean>
+
+    // We dont use color here because we want
+    // to be able to pass hex strings and rgb
+    // values to accent rather than a `Color`
+    accent?: SignalValue<PossibleColor>
+}
+```
+Next, create a component class for your components. It must extend an existing
+component such as `Line`, `Rect` or `Layout`. If you dont want to inherit any
+methods from an existing component, extend your class from `Node`. It is advisable
+to use a component most similar to yours, for instance if you were to make a component
+including a `Layout`, you should extend `Layout` and `LayoutProps`.
+```tsx
+export interface SwitchProps extends NodeProps { ... }
+
+export class Switch extends Node { ... }
+```
+
+To use the properties defined in the interface, your class _must_ contain a property with the
+same name. Motion Canvas provides type decorators to facilitate this. Here is an example of
+how you would define such a property:
+```tsx
+export class Switch extends Node {
+    // @initial - sets the property to an
+    // initial value if it was not provided.
+    @initial(false)
+    // @signal - is required by motion canvas
+    // for every prop that was passed in.
+    @signal()
+    public declare readonly initialState: SimpleSignal<boolean, this>
+
+    @initial("#68ABDF")
+    @signal()
+    public declare readonly accent: ColorSignal<this>
+```
```

Comment:
Yet another super minor detail: Personally I am missing a `}`. Or 
```ts
// ...
}
```
Since you move on to a different section in the next code snipped.

Reply:
okay. will be in the next commit in the name of consistency if its like this everywhere else